Humbertino_Avilez/CRUD-EXCEPTION/Vet_Patitas/src/main/java/com/bolsadeideas/springboot/bootcamp/app/controllers/AnimalController.java

```diff
@@ -0,0 +1,63 @@
+package com.bolsadeideas.springboot.bootcamp.app.controllers;
+
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.bolsadeideas.springboot.bootcamp.app.entity.Animal;
+import com.bolsadeideas.springboot.bootcamp.app.entity.ResponseAnimal;
+import com.bolsadeideas.springboot.bootcamp.app.service.InterfaceAnimalService;
+import com.bolsadeideas.springboot.bootcamp.app.utils.Constants;
+import static com.bolsadeideas.springboot.bootcamp.app.utils.Constants.ANIMAL_CREATED;
+import static com.bolsadeideas.springboot.bootcamp.app.utils.Constants.ANIMAL_FOUND;
+import static com.bolsadeideas.springboot.bootcamp.app.utils.Constants.ANIMAL_UPDATED;
+import static com.bolsadeideas.springboot.bootcamp.app.utils.Constants.ANIMAL_DELETED;
+
+@RestController
+public class AnimalController {
+
+	@Autowired
+	private InterfaceAnimalService animalService;
+
+	@GetMapping("/animals")
+	public List<Animal> toList() {
+		return animalService.findAll();
+	}
+
+	@GetMapping("/animals/{id}")
+	public ResponseEntity<ResponseAnimal<String>> detailsAnimal(@PathVariable Long id) {
+		Animal animal = animalService.findById(id);
```

Comment:
why put this in a variable? you don't use this variable in more that one plase. put directly "animalService.findById(id);" inside the new ResponseAnimal, change all the endpoints